webapp/views/App/views/Data/MapView/CoordinateAttributeDataLayer/CoordinateAttributePolygon.js

```diff
@@ -0,0 +1,245 @@
+  const offset_east_meters = Number(polygonProperties.offset_east) || 0
+  const numberOfPointsNorth = Number(polygonProperties.number_of_points_north) || 0
+  const numberOfPointsEast = Number(polygonProperties.number_of_points_east) || 0
+  const numberOfPointsCircle = Number(polygonProperties.number_of_points_circle) || 0
+  const controlPointOffsetNorh = Number(polygonProperties.controlpoint_offset_north) || 0
+  const controlPointOffsetEast = Number(polygonProperties.controlpoint_offset_east) || 0
+  const isCircle = polygonProperties.isCircle == 'true' ? true : false
+  const radius = Number(polygonProperties.radius) || 0
+
+  const len_lat = MetersToDegreesLatitude(len_lat_meters) //how many degrees lat length is
+  const len_lng = MetersToDegreesLongitude(len_lng_meters, latitude) // how many degrees lng length is (at the middle point)
+  const control_point_latlng = L.latLng(
+    latitude + MetersToDegreesLatitude(controlPointOffsetEast),
+    longitude + MetersToDegreesLongitude(controlPointOffsetNorh, latitude)
+  )
+
+  const middle_latlng = L.latLng(latitude, longitude)
+  const north_point = GeometryUtil.destination(middle_latlng, 0, len_lat_meters / 2 + offset_north_meters)
+  const south_point = GeometryUtil.destination(middle_latlng, 180, len_lat_meters / 2 - offset_north_meters)
+  const northeast_corner = GeometryUtil.destination(north_point, 90, len_lng_meters / 2 + offset_east_meters)
+  const southwest_corner = GeometryUtil.destination(south_point, 270, len_lng_meters / 2 - offset_east_meters)
+  const bounds = L.latLngBounds(southwest_corner, northeast_corner)
+
+  const pointSize = 10
+  const pointSizeLat = MetersToDegreesLatitude(pointSize)
+
+  const pointDistanceLat = len_lat / numberOfPointsNorth
+  const startLat = bounds.getSouthWest().lat + pointDistanceLat / 3
+  const startLon = bounds.getSouthWest().lng + len_lng / numberOfPointsEast / 3
+  const [colors, setColors] = useState([...Array(numberOfPointsEast)].map(() => Array(numberOfPointsNorth).fill(0))) // this generates zero matrix
+  const [circleColors, setCircleColors] = useState(Array(numberOfPointsCircle).fill(0))
+  let circleControlPointIndex = 0
+
+  const onClickControlPoint = (j, i) => {
+    const newArray = [...colors]
+    newArray[j][i] = (newArray[j][i] + 1) % _.size(PointColor)
+    setColors(newArray)
+  }
+
+  const onClickCircleControlPoint = (i) => {
+    const newArray = [...circleColors]
+    newArray[i] = (newArray[i] + 1) % _.size(PointColor)
+    setCircleColors(newArray)
+  }
+
+  const circleControlPoints = () => {
+    const middle = [latitude, longitude]
+    switch (numberOfPointsCircle) {
+      case 4:
+        return nCircles(middle, radius / 2, 4, 45)
+      case 5:
+        return fiveCircles(middle, radius)
+      case 10:
+        return tenCircles(middle, radius)
+      case 12:
+        return twelveCirles(middle, radius)
+      case 21:
+        return twentyOneCircles(middle, radius)
+    }
```

Comment:
I think that this can be simplified  maybe using just this function: nCircles

Reply:
The circle positions are different depending on the number of circles. If I use only one nCircles function then I'd need to have many if..else's there to handle all the different positions